src/hooks/student-list-section/useFilterData.tsx

```diff
@@ -0,0 +1,53 @@
+import { FilterConfig } from "@/types/students/students-list-types";
+import { useInstituteDetailsStore } from "@/stores/student-list/useInstituteDetailsStore";
+
+export const useFilterData = (currentSession?: string) => {
+    const batchNames = useGetBatchNames(currentSession);
+    const statuses = useGetStatuses();
+    const genders = useGetGenders();
+
+    const filterData: FilterConfig[] = [
+        {
+            id: "batch",
+            title: "Batch",
+            filterList: batchNames,
+        },
+        {
+            id: "statuses",
+            title: "Status",
+            filterList: statuses,
+        },
+        {
+            id: "gender",
+            title: "Gender",
+            filterList: genders,
+        },
+    ];

```

Comment:
This data should come from a ts file if this data comes from an API then integrate that API from somewhere else export that data and use it here then return filter data. If this data is static store it in a separate file then import from there.